### PR TITLE
feat: support ncclnet v3 and remove the dependency on nccl in the installation environment

### PR DESCRIPTION
**Pull request by shjwudp**

### CODE REVIEW

File: cc/v3/nccl_net_v3.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+typedef struct {
+  char* name;     // Used mostly for logging.
+  char* pciPath;  // Path to the PCI device in /sys.
+  uint64_t guid;  // Unique identifier for the NIC chip. Important for
+                  // cards with multiple PCI functions (Physical or virtual).
+  int ptrSupport; // NCCL_PTR_HOST or NCCL_PTR_HOST|NCCL_PTR_CUDA
+  int speed;      // Port speed in Mbps.
+  int port;       // Port number.
+  int maxComms;   // Maximum number of comms we can create
+}ncclNetProperties_v3_t;
```
Comment (liuhatry) [inline cc/v3/nccl_net_v3.h:22]: need a space

File: cc/v4/nccl_net_v4.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+typedef struct {
+  char* name;     // Used mostly for logging.
+  char* pciPath;  // Path to the PCI device in /sys.
+  uint64_t guid;  // Unique identifier for the NIC chip. Important for
+                  // cards with multiple PCI functions (Physical or virtual).
+  int ptrSupport; // NCCL_PTR_HOST or NCCL_PTR_HOST|NCCL_PTR_CUDA
+  int speed;      // Port speed in Mbps.
+  int port;       // Port number.
+  int maxComms;   // Maximum number of comms we can create
+}ncclNetProperties_v4_t;
```
Comment (liuhatry) [inline cc/v4/nccl_net_v4.h:21]: need a space

Comment (shjwudp) [reply cc/v4/nccl_net_v4.h:21]: Limited by the association with nccl, this symbol must be this name :(

File: cc/v3/nccl_net_v3.h (hunk 1 above)
Comment (shjwudp) [reply cc/v3/nccl_net_v3.h]: OK

File: cc/v4/nccl_net_v4.h (hunk 2 above)
Comment (shjwudp) [reply cc/v4/nccl_net_v4.h]: DONE

**Pull request merged**